compositor_pipeline/src/pipeline/decoder.rs

```diff
@@ -14,27 +18,29 @@ use crossbeam_channel::{bounded, Receiver, Sender};
 pub mod fdk_aac;
 mod ffmpeg_h264;
 mod opus;
+mod resampler;
 
-pub struct Decoder {
-    #[allow(dead_code)]
-    video: Option<VideoDecoder>,
-    #[allow(dead_code)]
-    audio: Option<AudioDecoder>,
+#[derive(Debug, thiserror::Error)]
+pub enum ResamplerInitError {
+    #[error(transparent)]
+    ResamplerInitError(#[from] rubato::ResamplerConstructionError),
 }
 
+pub struct Decoder;
```

Comment:
All those decoder changes are made, because `Decoder` doesn't have to be held anymore in memory. Before it was kept in the pipeline state, because we implemented threads closing with a custom `Drop` impl, but currently `PipelineEvent::EOS` does it.

Those changes aren't caused by resampling itself, but I wanted to get rid of this.